typings/windows-process-tree/promises.d.ts

```diff
@@ -0,0 +1,34 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+declare module 'windows-process-tree/promises' {
+  import {
+    IProcessCpuInfo,
+    IProcessInfo,
+    IProcessTreeNode,
+    ProcessDataFlag,
+  } from 'windows-process-tree';
```

Comment:
This adds a circular dependency which will probably cause problems later

Reply:
You're right. It's incorrect due to the strict top-level initialization of `promises.ts`. The _type_ imports are fine, but not the mutual module/function imports. Let me see what I can do.

Reply:
Ok, I looked into this and ended up basically just removing the `export * as promises from './promises'`. This means:

```ts
import * as wpc from 'windows-process-tree/promises';
// ^ works

import { promises } from 'windows-process-tree';
// ^ not exposed
```

It's not quite as convenient, but doesn't seem terrible to me since it could be added in the future without an API break. But to do so requires either moving the bulk of `index.ts` around or introducing a somewhat smelly lazy export.

Reply:
Thanks, that seems like a good approach